### PR TITLE
Switch to official external-dns helm chart

### PR DESCRIPTION
**Pull request by KashifSaadat**
- Includes fixes to cloudaccess module and removes duplication of CloudInfo and DNSZoneManagement roles.
- Update to EKS v1.26

### CODE REVIEW

File: modules/cloudaccess/main.tf [hunk 1]
```diff
@@ -1,10 +1,4 @@
...
-  resource_suffix = var.resource_suffix
-
-  # an AWS IAM OIDC "provider" is NOT required when the trust is to google, nor when the user
-  # tells us they're managing it themselves, only when provisioning for an Azure client + tenant ID
-  create_azure_trust  = var.wayfinder_identity_azure_client_id != "" && var.wayfinder_identity_azure_tenant_id != "" ? true : false
-  create_google_trust = var.wayfinder_identity_gcp_service_account != "" ? true : false
-
-  azure_oidc_issuer = local.create_azure_trust ? "https://sts.windows.net/${var.wayfinder_identity_azure_tenant_id}/" : ""
+  resource_suffix   = var.resource_suffix != "" ? "-${var.resource_suffix}" : ""
+  azure_oidc_issuer = "https://sts.windows.net/${var.wayfinder_identity_azure_tenant_id}/"
```
Comment (mrsheepuk) [inline modules/cloudaccess/main.tf:3]: will this work if wayfinder_identity_azure_tenant_id is unspecified? if not, this needs to be guarded with var.from_azure == true...

File: wayfinder-cloudaccess.tf [hunk 2]
```diff
@@ -1,120 +1,13 @@
-module "iam_assumable_role_dns_zone_manager" {
-  count   = var.enable_wf_cloudaccess ? 1 : 0
-  source  = "terraform-aws-modules/iam/aws//modules/iam-assumable-role"
-  version = "5.17.0"
+module "wayfinder_cloudaccess" {
```
Comment (mrsheepuk) [inline wayfinder-cloudaccess.tf:1]: this is missing a count var enable_Wf_cloudaccess check

File: modules/cloudaccess/main.tf (hunk 1 above)
Comment (KashifSaadat) [reply modules/cloudaccess/main.tf:3]: I'll add your check back in, as it's better to fail with creating an oidc issuer with an empty string (more obvious error).

File: modules/cloudaccess/examples/complete/aws-to-aws.tfvars.sample [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Resource suffix is optional, specify this if you need to keep role definitions unique amongst
+# multiple workspaces using the same AWS Account.
+resource_suffix                 = "-app1"
```
Comment (mrsheepuk) [inline modules/cloudaccess/examples/complete/aws-to-aws.tfvars.sample:3]: this isn't needed by default on AWS - only on azure where role definitions need to be tenant-wide unique

Comment (KashifSaadat) [reply modules/cloudaccess/examples/complete/aws-to-aws.tfvars.sample:3]: I'll drop it

**Review by mrsheepuk — APPROVED** [review]

**Pull request merged**